app/Http/Controllers/Product/ProductController.php

```diff
@@ -0,0 +1,260 @@
+        return view('bid.bid_current',['bid'=>$bid]);
+    }
+
+    /**
+     * @param $request{user_id, real_bid_amound}
+     * @return $current_price
+     */
+    public function postBid(Request $request, $id) {
+        $bid = $this->bidRepository->find($id);
+        
+        if($bid->status == Config::get('constants.bid_status.begining')){
+            if($request->real_bid_amount <=  ($bid->current_price)){
+                return false;
+            }elseif ($request->real_bid_amount >= $bid->cost_sell) {
+                $this->amountCostSell($bid, $request, $id);
+            }else{
+                if($request->real_bid_amount < $bid->current_highest_price){
+                    $this->amountLessHightest($bid, $request, $id);
+                }elseif($request->real_bid_amount == $bid->current_highest_price){
+                    $this->amountEqualHightest($bid, $request, $id);
+                }else{
+                    $this->amountBetterHightest($bid, $request, $id);
+                }
+            }
+            
+        }elseif ($bid->status == Config::get('constants.bid_status.no_user_bid')) {
+
+            $this->amountNoUserBid($bid, $request, $id);
+        }else
+            return false;
+        $respont = array(
+                'current_highest_bidder_id'=> $bid->current_highest_bidder_id,
+                'current_highest_bidder_name'=>$bid->current_highest_bidder_name,
+                'current_price'=> $bid->current_price,
+            );
+        return $respont;  
+    }
+
+    /**
+     * update bid in case no user bid
+     * @param $bid, $request, $id
+     * @return void
+     */
+    public function amountNoUserBid($bid, $request, $id){
+
+        try {
+            $bid_session_user_current = array(
+                'user_id' => $request->user_id,
+                'bid_id' => $id,
+                'real_bid_amount' => $request->real_bid_amount,
+                'bid_amount' => $request->real_bid_amount,
+                'bid_type' => $this->bid_type_manual,
+                'created_at'=> $this->time_now
+            );
+
+            DB::beginTransaction();
+            //$this->user_bidRepository->create($bid_session_user_current);
+            $bid->update([
+                'current_price' => $request->real_bid_amount,
+                'current_highest_price' => $request->real_bid_amount,
+                'current_highest_bidder_id' => $request->user_id,
+                'current_highest_bidder_name' => $request->user_name,
+                'status' => Config::get('constants.bid_status.begining'),
+            ]);
+            DB::commit();
+            
+        } catch (Exception $e) {
+            DB::rollback();
+            return false;
+        }
+    }
+
+    /**
+     * update bid in case amount input is less than current hightest 
+     * @param $bid, $request, $id
+     * @return void
+     */
+    public function amountLessHightest($bid, $request, $id){
```

Comment:
ở đây e đã truyền sang toàn bộ data bid info rồi thì em truyền thêm biến id để lấy bid_id làm gì nhỉ ???? thừa params rồi